compiler/rustc_borrowck/src/type_check/mod.rs

```diff
@@ -2242,6 +2242,13 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
                             }
                         }
                     }
+                    CastKind::Transmute => {
+                        span_mirbug!(
+                            self,
+                            rvalue,
+                            "Unexpected CastKind::Transmute, should only appear after lowering_intrinsics",
+                        );
+                    }
```

Comment:
These kinds of restrictions should be attached to mir [phases](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/mir/syntax/enum.RuntimePhase.html). It seems reasonable to allow this for all runtime mir (but you can just allow it for postcleanup too if you want, I don't care either way). Can you update this comment, add an associated check in the mir validator, and document this condition on both the linked phase docs and on the definition of `CastKind::Transmute`?

Reply:
Done.

(See <https://github.com/rust-lang/rust/pull/108442#discussion_r1117896836> above for how it ended up like this.)

Reply:
Looks good, thank you!